### PR TITLE
Note commands addet to CLI bot

### PR DESCRIPTION
**Pull request by VityukY**
add note command for CLI: 
 case "update-note":
case "replace-note":
case "remove-note":
case "find-note":
case "note-by-text":

### CODE REVIEW

File: bot_cli/__main__.py [hunk 1]
```diff
@@ -41,6 +44,16 @@ def main(address_book):
                 commands.add_contact(args)
             case "add-note":
                 commands.add_note(notebook, args)
+            case "update-note":
```
Comment (LadaM) [inline bot_cli/__main__.py:47]: We have `change-note` command in documentation for it to be consistent with the `change-phone` command that we already have

File: bot_cli/__main__.py [hunk 2]
```diff
@@ -36,6 +42,18 @@ def main(address_book):
                 print_info("How can I help you?")
             case "add-contact":
                 commands.add_contact(args)
+            case "add-note":
+                commands.add_note(notebook, args)
+            case "update-note":
+                commands.update_note(notebook, args)
+            case "replace-note":
+                commands.replace_note(notebook, args)
```
Comment (LadaM) [inline bot_cli/__main__.py:50]: The functionality of your `replace_note` is more like what I would imagine under `change_note`. `update_note` looks like `add_text` I would discuss it with the team today on the meeting. But we cannot keep it like this.

File: bot_cli/commands.py [hunk 3]
```diff
@@ -248,9 +270,76 @@ def birthdays(args):
...
+    note_id, _ = notebook.add_note(text)
+    print_success(f"Note with id {note_id} created successfully")
+
+
+# command, index, text = args[0], args[1], " ".join(args[2:])
+@note_error_handler
+def update_note(notebook: Notes, args):
+    index, text = args[0], " ".join(args[1:])
+    notebook.update_note(int(index), text)
+    print_success("Note successfully updated")
```
Comment (LadaM) [inline bot_cli/commands.py:301]: Thanks for using it

File: bot_cli/commands.py [hunk 4]
```diff
@@ -248,9 +270,76 @@ def birthdays(args):
...
+        raise ValueError(
+            f"Note cannot be empty and must be more than {MIN_NOTE_LEN} characters long"
+        )
+    note_id, _ = notebook.add_note(text)
+    print_success(f"Note with id {note_id} created successfully")
+
+
+# command, index, text = args[0], args[1], " ".join(args[2:])
+@note_error_handler
+def update_note(notebook: Notes, args):
```
Comment (LadaM) [inline bot_cli/commands.py:298]: I'm not sure we would want to keep it. If so, it should be named `add-text` or something like this

File: bot_cli/commands.py [hunk 5]
```diff
@@ -248,9 +270,76 @@ def birthdays(args):
...
+    index, text = args[0], " ".join(args[1:])
+    notebook.update_note(int(index), text)
+    print_success("Note successfully updated")
+
+
+@note_error_handler
+def replace_note(notebook: Notes, args):
```
Comment (LadaM) [inline bot_cli/commands.py:305]: Would like a name to be `change_note` to correspond to the command `change-note`

File: bot_cli/commands.py [hunk 6]
```diff
@@ -248,9 +270,76 @@ def birthdays(args):
...
+    note = notebook.find_note_by_index(int(index))
+    if note["Tags"]:
+        str_tags = " ".join(note["Tags"])
+        print_success(f"Note: {note['Note']}\n Tags:{str_tags}")
+    else:
+        print_success(f"Note: {note['Note']}")
+
+
+@note_error_handler
+def note_by_text(notebook: Notes, args):
```
Comment (LadaM) [inline bot_cli/commands.py:330]: The command should be `search-note` with search string and same name for the method

File: bot_cli/commands.py [hunk 7]
```diff
@@ -248,9 +270,76 @@ def birthdays(args):
...
+    # find_note_by_subtext
+    text = " ".join(args)
```
Comment (LadaM) [inline bot_cli/commands.py:332]: Add validation for the search string to not be empty and to be greater than MIN_SEARCH_STR_LEN in length

File: bot_cli/commands.py [hunk 8]
```diff
@@ -248,9 +270,76 @@ def birthdays(args):
...
+    notes = notebook.find_note_by_subtext(text)
+    if not notes:
```
Comment (LadaM) [inline bot_cli/commands.py:334]: Please extract the functionality from my `show_all_notes` function that represents the table and use it in both places.

**Review by LadaM — CHANGES_REQUESTED** [review]
Some updates are needed

File: bot_cli/commands.py (hunk 6 above)
Comment (VityukY) [reply bot_cli/commands.py:330]: done
cli - search-note
func -  search_note

File: bot_cli/commands.py (hunk 7 above)
Comment (VityukY) [reply bot_cli/commands.py:332]: if text.isspace() or len(text) < MIN_NOTE_LEN:
        raise ValueError(
            f"Note cannot be empty and must be more than {MIN_NOTE_LEN} characters long"
        )
        
        add this part

File: bot_cli/commands.py (hunk 5 above)
Comment (VityukY) [reply bot_cli/commands.py:305]: change everywhere replace_note -> change_note (cli, cls method,  and commadn)

File: bot_cli/commands.py (hunk 4 above)
Comment (VityukY) [reply bot_cli/commands.py:298]: deleted (I always update the notes as I work on them -  but okay )

File: bot_cli/__main__.py (hunk 2 above)
Comment (VityukY) [reply bot_cli/__main__.py:50]: delete

File: bot_cli/__main__.py (hunk 1 above)
Comment (VityukY) [reply bot_cli/__main__.py:47]: update delete
replace rename to change note

File: bot_cli/commands.py (hunk 8 above)
Comment (VityukY) [reply bot_cli/commands.py:443]: i will do this when will be finished merging with main branch, cause for now it's add more conflicts and mess

**Review by VityukY — COMMENTED** [review]
all changes accepted

**Review by LadaM — APPROVED** [review]

**Pull request merged**